src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Completion/RazorCompletionContext.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using Microsoft.AspNetCore.Razor.Language;
+
+namespace Microsoft.CodeAnalysis.Razor.Completion
+{
+    internal class RazorCompletionContext
+    {
+        public RazorCompletionContext(
+            RazorSyntaxTree syntaxTree,
+            TagHelperDocumentContext? tagHelperDocumentContext = null,
+            bool isIncompleteRequest = false)
```

Comment:
I wonder if instead of this we should have our own trigger abstraction? For instance, I'd imagine this could be translated to a concept of `CompletionRequested` which would equate to ctrl + space in the IDE. Other benefits of building our own abstraction would be eventually enabling our completion handlers to understand what triggered a completion (i.e. was it `<` or `@` etc.)